src/backend/distributed/executor/multi_utility.c

```diff
@@ -630,10 +630,22 @@ ProcessIndexStmt(IndexStmt *createIndexStatement, const char *createIndexCommand
 
 		if (isDistributedRelation)
 		{
+			Oid namespaceId = InvalidOid;
+			Oid indexRelationId = InvalidOid;
 			ErrorIfUnsupportedIndexStmt(createIndexStatement);
 
-			/* if it is supported, go ahead and execute the command */
-			ExecuteDistributedDDLCommand(relationId, createIndexCommand, isTopLevel);
+			/*
+			 * If the command is supported, and the index does not exist,
```

Comment:
This comment is actually more misleading the helpful. It comes right before whether you check if the index exists, and is _after_ the part that checks whether the command is supported. It looked like an outdated comment to me.

Can this be changed to _Skip sending distributed CREATE INDEX statements when an index of that name already exists_? That better states our logic (since AFAIK, the local command continues execution).